### ISSUE TITLE
[4.0] Invalid markup created by frontediting_modules layout 

### ISSUE THREAD

**Issue by janschoenherr** [joomla/joomla-cms#35363]
### Steps to reproduce the issue

Publish a menu module with layout `default` to the `topbar` position of the Cassiopeia template. Login to the Joomla frontend. Check the markup of the module in the frontend.

### Expected result

Valid markup.

### Actual result

```
<ul class="mod-menu mod-list nav foo-bar">
    <a class="btn btn-link jmodedit" href="#" target="_self" aria-describedby="tip-123">
    <span class="icon-edit" aria-hidden="true"></span><span class="visually-hidden">Edit</span></a>
    <div role="tooltip" id="tip-123">Edit Module<br>Test Menu<br>Position: below-top</div>
    <li class="nav-item item-101 default current active"></li>
</ul>
```

### System information (as much as possible)

Joomla 4.0.*

### Additional comments

The following line https://github.com/joomla/joomla-cms/blame/4.0-dev/layouts/joomla/edit/frontediting_modules.php#L40 will create invalid markup, if the module's outer tag is an `<ul>` or `<ol>`. (e.g.

The Unordered List element `<ul` may only contain `<li>`, `<template>`, `<script>` child elements.

Introduced with https://github.com/joomla/joomla-cms/commit/678e929f777e326c81cc12a015418c13d9199d50

**Comment by brianteeman** [comment]
I can not replicate this - maybe it was changed since the issue was first created

![image](https://user-images.githubusercontent.com/1296369/155561287-b72a0df8-4bf4-4aa1-9822-211725dd71f4.png)

**Comment by janschoenherr** [comment]
@brianteeman I just updated the Steps to reproduce. Maybe it needs to be the `topbar` position, because the module does not get wrapped there.

**Comment by brianteeman** [comment]
Yes that was it. It is a bug but very low priority. Do you have an idea how to fix it

**Comment by janschoenherr** [comment]
I can only think of wrapping the modules in an extra `<div>`.

Trying to insert the button into the first `<li>` can possibly fail if there is none.

**Comment by brianteeman** [comment]
Can't really see the point myself

**Comment by brianteeman** [comment]
Can someone remove the information required tag please
